Drivers/Src/CANbus.c

```diff
@@ -79,16 +115,17 @@ ErrorStatus CANbus_Send(CANId_t id, CANPayload_t payload, CAN_blocking_t blockin
     else
     {
         OSSemPend(
-            &CANMail_Sem4,
+            &(CANMail_Sem4[bus]),
             0,
             OS_OPT_PEND_NON_BLOCKING,
             &timestamp,
             &err);
 
         // don't crash if we are just using this in non-blocking mode and don't block
-        if (err != OS_ERR_PEND_WOULD_BLOCK) {
-            assertOSError(OS_CANDRIVER_LOC,err);
+        if(err == OS_ERR_PEND_WOULD_BLOCK){
```

Comment:
Should the function return if in non-blocking mode and unable to pend on the semaphore?